pkg/filestore/filestore.go

```diff
@@ -18,29 +18,30 @@ import (
 	"errors"
 	"fmt"
 	"io"
+	"io/fs"
 	"os"
 	"path/filepath"
 
 	"github.com/tus/tusd/v2/internal/uid"
 	"github.com/tus/tusd/v2/pkg/handler"
 )
 
-var defaultFilePerm = os.FileMode(0664)
-var defaultDirectoryPerm = os.FileMode(0754)
-
 // See the handler.DataStore interface for documentation about the different
 // methods.
 type FileStore struct {
 	// Relative or absolute path to store files in. FileStore does not check
 	// whether the path exists, use os.MkdirAll in this case on your own.
 	Path string
+
+	DirPerm  fs.FileMode
+	FilePerm fs.FileMode
 }
 
 // New creates a new file based storage backend. The directory specified will
 // be used as the only storage entry. This method does not check
 // whether the path exists, use os.MkdirAll to ensure.
-func New(path string) FileStore {
-	return FileStore{path}
+func New(path string, dirPerms uint32, filePerms uint32) FileStore {
```

Comment:
We shouldn't change the signature of exported functions as this would be a breaking change. Instead, we can keep `New` unchanged and users can set the `DirPerm` and `FilePerm` fields after they called `New`.

Reply:
I corrected the code, please take a look.